### PR TITLE
No Bug: Add BT compile time flag check

### PR DESCRIPTION
**Pull request by kylehickinson**
## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -12078,6 +12081,7 @@
 					"-ObjC",
 					"-lxml2",
 				);
+				OTHER_SWIFT_FLAGS_BASE = "-DMOZ_TARGET_$(TARGET_NAME:upper) -DNO_USER_WALLETS -DNO_SKUS -DNO_BRAVE_TODAY";
```
Comment (iccub) [inline Client.xcodeproj/project.pbxproj]: why not placing it in base.xconfig or other xconfig files?

Comment (kylehickinson) [reply Client.xcodeproj/project.pbxproj]: Good point, I completely forgot the others were in Base.xcconfig, will fix

File: Client/Configuration/Base.xcconfig [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #include "Local/BuildId.xcconfig"
 #include "Local/Version.xcconfig"
 
-OTHER_SWIFT_FLAGS_BASE=-DMOZ_TARGET_$(TARGET_NAME:upper) -DNO_USER_WALLETS -DNO_SKUS
+OTHER_SWIFT_FLAGS_BASE=-DMOZ_TARGET_$(TARGET_NAME:upper) -DNO_USER_WALLETS -DNO_SKUS -DNO_BRAVE_TODAY
```
Comment (iccub) [inline Client/Configuration/Base.xcconfig:13]: all these flags are your work :D r.i.p

**Review by iccub — APPROVED** [review]

File: Client/Configuration/Base.xcconfig (hunk 2 above)
Comment (kylehickinson) [reply Client/Configuration/Base.xcconfig:13]: I know 😭

**Pull request merged**